File: solutions/_367_valid_perfect_square/solution.java

package _367_valid_perfect_square;

class Solution {
    public boolean isPerfectSquare(int num) {
        long L = 1;
        long R = num;
        while (L <= R) {
            long mid = L + (R - L) / 2;
            long guess = mid * mid;
            if (guess > num)
                R = mid - 1;
            else if (guess < num)
                L = mid + 1;
            else
                return true;

        }
        return false;
    }
}

/*
 * https://leetcode.com/problems/valid-perfect-square/
 */

File: solutions/_367_valid_perfect_square/solution.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        L = 1
        R = num
        while L <= R:
            mid = (L + R) // 2
            guess = mid * mid
            if guess > num:
                R = mid - 1
            elif guess < num:
                L = mid + 1
            else:
                return True
        return False


# https://leetcode.com/problems/valid-perfect-square/
